wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -1069,6 +1097,56 @@ public void onNewKeyChainEncrypted() {
 
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+        findViewById(R.id.restart_sync_icon).setOnClickListener(new View.OnClickListener() {
+            public void onClick(final View v) {
+                findViewById(R.id.sync_error_pane).setVisibility(View.GONE);
+                findViewById(R.id.sync_progress_pane).setVisibility(View.VISIBLE);
+            }
```

Comment:
The restart button appears to only change the status visually.  Does it try to reconnect?  

I am not sure how to get the block chain service to restart.